src/core/optional.h

```diff
@@ -6,8 +6,8 @@ template <typename T> class optional {
   bool is_some;
 
 public:
-  optional(T value) : value{value}, is_some{true} {}
-  optional() : value{T()}, is_some{false} {}
+  optional(const T &value) : value(value), is_some(true) {}
+  optional() : value(T()), is_some(false) {}
```

Comment:
@nathansobo Using the curly brace style of initialization here was causing build errors unless we used the latest clang or gcc. I changed it to use direct initialization so that downstream projects wouldn't have to change their CI config.

```
../src/core/optional.h:9:36: error: no viable conversion from 'const Patch::Hunk' to 'Point'
  optional(const T &value) : value{value}, is_some{true} {}
```

It seems like list initialization is only supported by fairly new compilers, so clang is trying to perform aggregate initialization instead. I guess this means that we're allowing narrowing conversions here now, but I think it's worth it to allow this project to be built with less shenanigans on travis and appveyor.